model/task_queue_service.go

```diff
@@ -85,27 +87,33 @@ func (s *taskDispatchService) Refresh(distroID string) error {
 }
 
 func (s *taskDispatchService) ensureQueue(distroID string) (CachedDispatcher, error) {
-	// If there is a "distro": *basicCachedDispatcherImpl in the cachedDispatchers map, return that.
-	// Otherwise, get the "distro"'s taskQueue from the database; seed its cachedDispatcher; put that in the map and return it.
-	s.mu.RLock()
+	d, err := distro.FindOne(distro.ById(distroID))
+	if err != nil {
+		return nil, errors.Wrapf(err, "Database error for find() by distro id '%s'", distroID)
+	}
 
-	distroDispatchService, ok := s.cachedDispatchers[distroID]
-	s.mu.RUnlock()
-	if ok {
-		return distroDispatchService, nil
+	config, err := evergreen.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "error retrieving Evergreen config")
+	}
+	plannerSettings, err := d.GetResolvedPlannerSettings(config)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error resolving the PlannerSettings for distro '%s'", d.Id)
 	}
 
+	// If there is a "distro": *basicCachedDispatcherImpl in the cachedDispatchers map, return that.
+	// Otherwise, get the "distro"'s taskQueue from the database; seed its cachedDispatcher; put that in the map and return it.
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	distroDispatchService, ok = s.cachedDispatchers[distroID]
+
+	distroDispatchService, ok := s.cachedDispatchers[distroID]
 	if ok {
-		return distroDispatchService, nil
+		if time.Since(distroDispatchService.CreatedAt()) < s.ttl && distroDispatchService.Type() == plannerSettings.Version {
```

Comment:
If the type matches and it's greater than the ttl, don't we still want to continue?